app/middleware/get-service-and-gateway-account.middleware.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const _ = require('lodash')
+
+const logger = require('../utils/logger')(__filename)
+const paths = require('../../app/paths')
+const { NotFoundError } = require('../../app/errors')
+const Connector = require('../services/clients/connector.client.js').ConnectorClient
+
+const { keys } = require('@govuk-pay/pay-js-commons').logging
+const connectorClient = new Connector(process.env.CONNECTOR_URL)
+
+async function getGatewayAccountByExternalId (gatewayAccountExternalId, correlationId) {
+  if (!gatewayAccountExternalId) {
```

Comment:
I believe it's an OK scenario if the gateway account hasn't been explicitly specified on the path, this might be the case if the route is at the service level (organisation settings for example).

I think this will throw a 500 internal server error if that's ever not defined on the path. As this middleware is just focussed on fetching ideally the only error it would throw is if it was asked to get something and that wasn't found (if there's no ID it could instead not try and get it).

Reply:
👍 Agree, changed to not throw error when gateway account external ID is not available. Also made it mandatory for at least one external ID (gateway account or service) to be available on the request and should have `service` set on the request based on available external ID